galsim/table.py

```diff
@@ -195,20 +195,20 @@ def __call__(self, x):
             if dimen > 2:
                 raise ValueError("Arrays with dimension larger than 2 not allowed!")
             elif dimen == 2:
-                f = np.empty_like(x.ravel()).astype(float)
-                self.table.interpMany(x.astype(float).ravel(),f)
+                f = np.empty_like(x.ravel(), dtype=float)
```

Comment:
Whenever you do these ```empty_like``` (or ```zeros_like``` or ```ones_like```) I thought the rule is that it gives it the type of the argument.  Is that right?  If so, then given that you've already enforced ```x``` and ```f``` to have type float, perhaps all these ```dtype``` kwargs (here and later) are unnecessary?

I just did a quick check and found that having the ```dtype``` kwarg seemed to increase the time by 60%, so if it's unnecessary I imagine there are a bunch of places throughout the code where it could helpfully be removed.  (If you want I can post a code snippet...)